src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -156,21 +159,133 @@ private boolean match(VendorData vendorData) {
+   * @return True if subsequence score is not lesser than the threshold, Otherwise False.
+   * @see <a href="https://www.geeksforgeeks.org/longest-common-subsequence-dp-4/">Longest Common
+   *      Subsequence Problem</a>
+   */
+  private static boolean longestCommonSubsequenceCheck(String one, String two) {
+    return LCS.apply(one, two) >= (one.length() * METRIC_THRESHOLD);
+  }
+
+  /**
+   * Find the Jaro Winkler Similarity score between the 2 strings. And check if the score is not
+   * lesser than a certain threshold.
+   * 
+   * @param String one.
+   * @param String two.
+   * @return True if the score is not lesser than the threshold, Otherwise False.
+   * @see <a href="https://www.geeksforgeeks.org/jaro-and-jaro-winkler-similarity/">Jaro Winkler
+   *      Similarity</a>
+   */
+  private static boolean jaroWinklerSimilarityCheck(String one, String two) {
+    return JWS.apply(one, two) > METRIC_THRESHOLD;
+  }
+
+  /**
+   * Checks if one of the cpeXXUri contains the project's name.
+   * 
+   * @param cpeMatch of type {@link CpeMatch}.
+   * @return True if one of the cpeXXUri contains the project's name. Otherwise False.
    */
-  private static boolean match(String one, String two) {
-    one = one.toLowerCase().trim();
-    two = two.toLowerCase().trim();
-    return one.contains(two) || two.contains(one);
+  private boolean cpeUriCheck(CpeMatch cpeMatch) {
+    return match(productFrom23Uri(cpeMatch.getCpe23Uri()))
+        || match(productFrom22Uri(cpeMatch.getCpe22Uri()));
+  }
+
+  /**
+   * Returns product name from {@link CpeMatch#getCpe23Uri()}.
+   */
+  private static String productFrom23Uri(String cpe23Uri) {
+    return stringAtIndex(cpe23Uri, ":", 4);
+  }
+
+  /**
+   * Returns product name from {@link CpeMatch#getCpe22Uri()}.
+   */
+  private static String productFrom22Uri(String cpe22Uri) {
+    return stringAtIndex(cpe22Uri, ":", 3);
+  }
+
+  /**
+   * Returns the substring present at the index after the split by given delimiter pattern.
+   * 
+   * @param content The String to be split.
+   * @param pattern The delimiter to be used.
+   * @param index The index of the sub-string which should be returned.
+   * @return The sub-string if index is valid, else returns an empty string.
+   */
+  private static String stringAtIndex(String content, String pattern, int index) {
+    if (content != null) {
+      String[] splitContent = content.split(pattern);
+      if (splitContent.length > index) {
+        return splitContent[index];
+      }
+    }
+
+    return StringUtils.EMPTY;
+  }
+
+
+  /**
+   * This is recursive method to parse through the list of nodes. Also its children nodes if
+   * present. To find a match to the project's name.
+   * 
+   * @param nodes List of type {@link Node}.
+   * @return True if there was a match. Otherwise False.
+   */
+  private boolean parseNodes(List<Node> nodes) {
+    for (Node node : nodes) {
+
+      if (node.getChildren() != null && !node.getChildren().isEmpty()
```

Comment:
Could you please simplify this by moving the null check to the beginning of the method? The, `isEmpty()` check is not necessary any more.

```java
private boolean parseNodes(List<Node> nodes) {
  if (nodes == null) {
      return false;
  }

  for (Node node : nodes) {

    if (parseNodes(node.getChildren()) {
      return true;
```

Reply:
Good idea. This would be better.